### PR TITLE
Allow transformed classes to be dumped

### LINKED ISSUE #11857: quarkus.debug.transformed-classes-dir does not work anymore
This regression was introduced in:
https://github.com/quarkusio/quarkus/commit/9d3fa16a74c1b9354352843aab28ce71b3c95350#diff-db1d39a812769ff88e14c8dd8e39c0c4

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Fixes #11857

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/steps/ClassTransformingBuildStep.java [hunk 1]
```diff
@@ -104,7 +107,22 @@ TransformedClassesBuildItem handleClassTransformation(List<BytecodeTransformerBu
...
                                 }
                                 cr.accept(visitor, 0);
-                                return new TransformedClassesBuildItem.TransformedClass(className, writer.toByteArray(),
+                                byte[] data = writer.toByteArray();
+                                if (BootstrapDebug.DEBUG_TRANSFORMED_CLASSES_DIR != null) {
+                                    File debugPath = new File(BootstrapDebug.DEBUG_TRANSFORMED_CLASSES_DIR);
+                                    if (!debugPath.exists()) {
+                                        debugPath.mkdir();
+                                    }
+                                    File classFile = new File(debugPath, className.replace(".", "/") + ".class");
```
Comment (jaikiran) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/ClassTransformingBuildStep.java:116]: I always forget - does this scheme of using `/` character as a file path seperator in the `File` constructor work fine on Windows?

**Comment by geoand** [comment]
@mkouba please give it a try as you were the one that opened the original issue

File: core/deployment/src/main/java/io/quarkus/deployment/steps/ClassTransformingBuildStep.java (hunk 1 above)
Comment (stuartwdouglas) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/ClassTransformingBuildStep.java:116]: yes

**Review by mkouba — APPROVED** [review]
I confirm that it works fine again!

**Pull request merged**